crates/contract/src/update.rs

```diff
@@ -461,6 +462,73 @@ mod tests {
         assert_eq!(found, expected);
     }
 
+    /// Asserts that [`ProposedUpdates::do_update`] clears all entries and votes.
+    #[test]
+    fn test_proposed_updates_do_update_clears_all_state() {
+        // Given: multiple update proposals with votes from different accounts
+        let mut proposed_updates = ProposedUpdates::default();
+
+        let update_0 = Update::Contract([0; 1000].into());
+        let update_id_0 = proposed_updates.propose(update_0.clone());
+
+        let update_1 = Update::Contract([1; 1000].into());
+        let update_id_1 = proposed_updates.propose(update_1.clone());
+
+        let update_2 = Update::Config(Config::default());
+        let update_id_2 = proposed_updates.propose(update_2.clone());
+
+        let account_0 = gen_account_id();
+        let account_1 = gen_account_id();
+        let account_2 = gen_account_id();
+
+        proposed_updates.vote(&update_id_0, account_0.clone());
+        proposed_updates.vote(&update_id_1, account_1.clone());
+        proposed_updates.vote(&update_id_2, account_2.clone());
+
+        let before: TestUpdateVotes = (&proposed_updates).try_into().unwrap();
+        let expected_before = TestUpdateVotes {
+            id: 3,
+            entries: BTreeMap::from([
+                (
+                    0,
+                    UpdateEntry {
+                        update: update_0.clone(),
+                        votes: HashSet::from([account_0.clone()]),
+                        bytes_used: bytes_used(&update_0),
+                    },
+                ),
+                (
+                    1,
+                    UpdateEntry {
+                        update: update_1.clone(),
+                        votes: HashSet::from([account_1.clone()]),
+                        bytes_used: bytes_used(&update_1),
+                    },
+                ),
+                (
+                    2,
+                    UpdateEntry {
+                        update: update_2.clone(),
+                        votes: HashSet::from([account_2.clone()]),
+                        bytes_used: bytes_used(&update_2),
+                    },
+                ),
+            ]),
+        };
+        assert_eq!(before, expected_before);
+
+        // When: executing an update
+        proposed_updates.do_update(&update_id_1, UPDATE_CONFIG_GAS);
+
+        // Then: all state is cleared (entries and votes)
+        let after: TestUpdateVotes = (&proposed_updates).try_into().unwrap();
+        let expected_after = TestUpdateVotes {
+            id: 3,
```

Comment:
Oh, I didn't now we keep the id. I guess this makes sense, as it avoids race conditions.
Did you check if this persists after migration & could you add a sentence or two about this to the contract readme?